src/main/java/com/google/swarm/tokenization/txt/ConvertTxtToDLPRow.java

```
@@ -47,36 +47,40 @@ public ConvertTxtToDLPRow(
-        values.forEach(
-            value -> rowBuilder.addValues(Value.newBuilder().setStringValue(value).build()));
-        context.output(KV.of(fileName, rowBuilder.build()));
-
-      } else {
-        LOG.warn(
-            "Rows must have the same number of items {} as there are headers {}",
-            values.size(),
-            csvHeader.size());
-      }
-    } else {
-      rowBuilder.addValues(Value.newBuilder().setStringValue(input).build());
-      context.output(KV.of(fileName, rowBuilder.build()));
+    try {
+        Table.Row.Builder rowBuilder = Table.Row.newBuilder();
+        String input = Objects.requireNonNull(context.element().getValue());
+        Map<String, List<String>> headers = context.sideInput(header);
+        String fileName = context.element().getKey();
+        List<String> csvHeader = headers.get(fileName);
+        if (csvHeader == null) {
+          throw new RuntimeException(
+              "Unable to find header row for fileName: "
+                  + fileName
+                  + ". The side input only contains header for "
+                  + headers.keySet());
+        }
+    
+        if (columnDelimiter != null) {
+    
+          List<String> values = Util.parseLine(input, columnDelimiter, '"');
+          if (values.size() == csvHeader.size()) {
+            values.forEach(
+                value -> rowBuilder.addValues(Value.newBuilder().setStringValue(value).build()));
+            context.output(KV.of(fileName, rowBuilder.build()));
+    
+          } else {
+            LOG.warn(
+                "Rows must have the same number of items {} as there are headers {}",
+                values.size(),
+                csvHeader.size());
+          }
+        } else {
+          rowBuilder.addValues(Value.newBuilder().setStringValue(input).build());
+          context.output(KV.of(fileName, rowBuilder.build()));
+        }
+    } catch (IllegalArgumentException e) {
+        LOG.info(e.getMessage());
     }
```

When can this happen? What is the impact of logging and swallowing the exception on the pipeine?

---

Fixed